### PR TITLE
Add deserialization_exception field for programmatic error handling

### PR DESCRIPTION
**Pull request by bjester**
## Summary
This allows programmatic handling of deserialization errors instead of parsing error messages

- Adds a new `deserialization_exception` nullable field to `Store` model that stores the fully-qualified exception class path when deserialization fails
- Updates existing `deserialization_error` field to be nullable
- Updates handling of deserialization errors to set both fields, or otherwise null both of them
- Adds tests for changes
- Removes migration tests that verified the squash migration, which now cause issues with new migrations

## References
Followup to #306
Closes https://github.com/learningequality/morango/issues/315

## AI Usage
Directed an AI agent to tackle the issue, iterated on its work for a loop, then did a small refactor myself. Had the AI agent add test coverage, then reviewed.

### CODE REVIEW

File: morango/sync/operations.py [hunk 1]
```diff
@@ -512,7 +517,20 @@ def _deserialize_from_store(profile, skip_erroring=False, filter=None):
 
             # if requested, skip any records that previously errored, to be faster
             if skip_erroring:
-                store_models = store_models.filter(deserialization_error="")
+                # previously, deserialization_error was not nullable, and set to empty string if
+                # there wasn't an error, so this nullifies empty strings to ensure the filter
+                # applies either way
+                store_models = (
```
Comment (ozer550) [inline morango/sync/operations.py:523]: Doesn't this keep the errored rows instead of skipping them? as mentioned in the comment above?
`# if requested, skip any records that previously errored, to be faster`

Comment (ozer550) [reply morango/sync/operations.py:523]: Can we also add a test to verify this `skip_erroring` behavior after the null string change?

**Review by ozer550 — CHANGES_REQUESTED** [review]
I might be wrong but it seems like we are doing opposite of "skip rows that previously failed deserialization" ? Was this change in behavior intended? Also adding some tests to verify the null string change for skipping would be useful!

File: morango/sync/operations.py (hunk 1 above)
Comment (bjester) [reply morango/sync/operations.py:523]: Yeah I think you're right, it should be `__isnull=True`. Good catch!

Comment (bjester) [reply morango/sync/operations.py:523]: Fixed! And added tests.

File: morango/sync/operations.py
Comment (ozer550) [inline morango/sync/operations.py]: Doing a grep on `deserialization_error`, would we want the `deserialization_exception` populated here as well?

**Review by ozer550 — COMMENTED** [review]
I did a grep on places where we had "deserialization_exception" and found a place where we might need to add the execution path as well?

File: morango/sync/operations.py
Comment (bjester) [reply morango/sync/operations.py]: I think you're right, we probably should

**Comment by bjester** [comment]
@ozer550 I've updated it to set the exception in the code you found. I also grepped and found some backend code setting to empty string. We probably didn't to set it there, but I updated the code anyway, to be have parity with `deserialization_error` handling.

File: morango/sync/operations.py [hunk 2]
```diff
@@ -310,33 +316,32 @@ def _validate_missing_store_foreign_keys(from_model_name, to_model_name, temp_ta
...
         c.execute(select_sql)
         for from_field, from_pk, to_pk in c.fetchall():
-            err = dict(
-                {
-                    from_field: "{to_model_name} instance with id '{to_pk}' does not exist".format(
-                        to_model_name=to_model_name,
-                        to_pk=to_pk_field.to_python(to_pk),
-                    )
-                }
+            err = IntegrityError(
```
Comment (ozer550) [inline morango/sync/operations.py:330]: Is there a reason we are specifically choosing IntegrityError here? This is validation issue rather than DB integrity one? Might lead to misinterpretations? Would it be better to have a special dedicated Morango error class?

**Review by ozer550 — COMMENTED** [review]
Just a small suggestion not a blocker regarding the Interity Error. Otherwise looks good to me.

File: morango/sync/operations.py (hunk 2 above)
Comment (bjester) [reply morango/sync/operations.py:330]: Nominally the function "validates", but it validates FKs. The previous implementation was trying to emulate a validation error, but in reality, a missing FK is an integrity issue. Continuing to have it be represented as a validation error, which isn't entirely specific, doesn't benefit the ability to analyze the errors (i.e. new `deserialization_exception`). So this is an improvement by being more specific, which wasn't really necessary before without `deserialization_exception`

**Review by ozer550 — APPROVED** [review]
Lgtm! Thanks Blaine.
